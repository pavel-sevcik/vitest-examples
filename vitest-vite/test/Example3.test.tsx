import { expect, test } from 'vitest'

import renderer from 'react-test-renderer'

import Link from '@/components/Link'

function toJson(component: renderer.ReactTestRenderer) {
    const result = component.toJSON()

    expect(result).toBeDefined()
    expect(result).not.toBeInstanceOf(Array)

    return result as renderer.ReactTestRendererJSON
}

test('Link changes the class when hovered', () => {
    const component = renderer.create(
        <Link page="https://google.com">Google</Link>,
    )

    let tree = toJson(component)
    expect(tree).toMatchSnapshot()

    tree.props.onMouseEnter()

    tree = toJson(component)
    expect(tree).toMatchSnapshot()

    tree.props.onMouseLeave()

    tree = toJson(component)
    expect(tree).toMatchSnapshot()
})
