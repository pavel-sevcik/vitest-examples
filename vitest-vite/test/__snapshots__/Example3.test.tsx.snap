// Vitest Snapshot v1

exports[`Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="https://google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Google
</a>
`;

exports[`Link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="https://google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Google
</a>
`;

exports[`Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="https://google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Google
</a>
`;
